src/chains/ethereum/transaction/src/legacy-transaction.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  Data,
+  Quantity,
+  keccak,
+  BUFFER_EMPTY,
+  BUFFER_32_ZERO,
+  RPCQUANTITY_EMPTY
+} from "@ganache/utils";
+import { Address } from "@ganache/ethereum-address";
+import type Common from "@ethereumjs/common";
+import { ecsign } from "ethereumjs-util";
+import { encodeRange, digest } from "@ganache/rlp";
+import { BN } from "ethereumjs-util";
+import {
+  hasPartialSignature,
+  RuntimeTransaction,
+  toValidLengthAddress
+} from "./runtime-transaction";
+import { TypedRpcTransaction } from "./rpc-transaction";
+import {
+  LegacyDatabasePayload,
+  LegacyDatabaseTx,
+  TypedDatabaseTransaction
+} from "./raw";
+import { computeInstrinsicsLegacyTx } from "./signing";
+
+export class LegacyTransaction extends RuntimeTransaction {
+  public gasPrice: Quantity;
+  public type: Quantity = Quantity.from("0x0");
+
+  public constructor(
+    data: LegacyDatabasePayload | TypedRpcTransaction,
+    common: Common
+  ) {
+    super(data, common);
+    if (Array.isArray(data)) {
+      this.nonce = Quantity.from(data[0], true);
```

Comment:
I think this should be:

```suggestion
      this.nonce = Quantity.from(data[0]);
```